namerd/main/src/main/scala/io/buoyant/namerd/DtabHandler.scala

```diff
@@ -0,0 +1,65 @@
+package io.buoyant.namerd
+
+import com.twitter.finagle.http.{MediaType, Status, Response, Request}
+import com.twitter.finagle.{Service, Dtab}
+import com.twitter.util.Future
+import io.buoyant.linkerd.admin.names.DelegateApiHandler
+
+class DtabHandler(
+  store: DtabStore
+) extends Service[Request, Response] {
+
+  import DtabHandler._
+
+  override def apply(req: Request): Future[Response] = req.path match {
+    case rexp(namespace) =>
+      store.observe(namespace).sample() match {
```

Comment:
Depending on the implementation, this can create an unobserved `Activity` and sampling it may not get the correct value.  A safer thing would be `store.observer(namespace).values.toFuture.flatMap(Future.const).map { case Some(dtab) => ...`